File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/PhoneNumber.java
package kz.greetgo.sandbox.controller.model;

public class PhoneNumber {
    private PhoneType phoneType;
    private String number;

    public PhoneNumber(PhoneType phoneType, String number) {
        this.phoneType = phoneType;
        this.number = number;
    }

    public PhoneType getPhoneType() {
        return phoneType;
    }

    public void setPhoneType(PhoneType phoneType) {
        this.phoneType = phoneType;
    }

    public String getNumber() {
        return number;
    }

    public void setNumber(String number) {
        this.number = number;
    }
}

File: sandbox.client/front/ts/model/PhoneNumber.ts

import {PhoneType} from "./PhoneType";

export class PhoneNumber {
    public phoneType: PhoneType;
    public number: number;

    public assign(o: any): PhoneNumber {
        this.phoneType = o.phoneType;
        this.number = o.number;

        return this;
    }

    public static copy(a: any): PhoneNumber {
        let ret = new PhoneNumber();
        ret.assign(a);
        return ret;
    }
}
